src/DataStructures/DataVector.hpp

```diff
@@ -481,12 +481,11 @@ namespace MakeWithValueImpls {
 /// \brief Returns a DataVector the same size as `input`, with each element
 /// equal to `value`.
 template <>
-struct MakeWithValueImpl<DataVector, DataVector> {
-  static SPECTRE_ALWAYS_INLINE DataVector apply(const DataVector& input,
-                                                const double value) {
-    return DataVector(input.size(), value);
-  }
-};
+SPECTRE_ALWAYS_INLINE DataVector
+MakeWithValueImpl<DataVector, DataVector>::apply(const DataVector& input,
+                                                 const double value) {
```

Comment:
You've lost a lot of `noexcept`s in this commit

Reply:
This just reverts the `MakeWithValueImpl` to the state in develop, which had many missing `noexcept`s. With the MakeNaN utility, this PR no longer touches `MakeWithValue` at all

Reply:
Okay, I'll revisit this at the end when your additions are clearer :)

Reply:
would it be preferable to squash this part so in vanishes from the fixup diffs?

Reply:
I wouldn't object to that, personally. I would find it easier to review, though I of course can't speak for other people :)